app/main/views/login.py

```diff
@@ -251,11 +253,13 @@ def create_user(encoded_token):
         # valid supplier account exists
         if user.role == 'supplier' and not is_registered_to_another_supplier and not is_inactive_or_locked:
             # valid supplier account exists and is logged in
-            if not current_user.is_anonymous() and current_user.email_address == user.email_address:
+            if not current_user.is_authenticated():
+                return redirect(url_for('.render_login'))
+            else:
+                if current_user.email_address != user.email_address:
+                    flash('You are trying to create a user while logged in as a different user', 'error')
```

Comment:
Not sure if this message still makes sense once you're redirected to the dashboard. But I suppose it's better than nothing and this shouldn't happen to begin with.